medcat/utils/decorators.py

```diff
@@ -7,16 +7,21 @@ def _format_version(ver: Tuple[int, int, int]) -> str:
     return ".".join(str(v) for v in ver)
 
 
-def deprecated(message: str, depr_version: Tuple[int, int, int], removal_version: Tuple[int, int, int]) -> Callable:
+def deprecated(message: str, depr_version: Tuple[int, int, int],
+               removal_version: Tuple[int, int, int],
+               allow_usage: bool = False) -> Callable:
```

Comment:
The method does indeed get monkey-patched during testing.

I suppose we could add `**kwargs` to the method to avoid describing the argument. But we can't remove it entirely since then the addition of the `allow_usage` keyword/argument would cause an exception during use (`TypeError` due to unexpected keyword argument).
But at the same time, we if this was hidden, the future maintainer of this resource might not know how to achieve the same result.

The main reason the method gets monkey-patched during testing is because we (generally) want to avoid using deprecated methods in our code. And if we had 100% test coverage (which we certainly don't - but that's besides the point), raising an exception during test time would guarantee that we're not.
But I've made an exception to this one, mostly so we can pre-specify when we'd stop supporting pydantic 1. So that we don't forget to do this at this later date, and so that it's actually documented (at least in code) and caught during GHA workflow (at the appropriate release time).

But if you've got some ideas on how to do this more elegantly, then don't hesitate to propose them!

Reply:
Alright. I think there is no need to deprecate `get_model_dump` and `get_model_fields` in this PR given the goal is to support both versions ATM and they are supposed to do the job well. Besides, it is a bit weird to add a new method and deprecate it at the same time but no strong opinion on this.

Also, there will be UserWarnings like `Field "model_X" has conflict with protected namespace "model_"` so to me, extra work including renaming needs to be done before medcat fully embraces pydantic v2 and deprecates v1.